### PR TITLE
New resource: azurerm_ip_groups

### PR DESCRIPTION
**Pull request by manicminer**
- `azurerm_firewall_network_rule_collection`: support destination_ip_groups and source_ip_groups for rules
- `azurerm_firewall_nat_rule_collection`: support source_ip_groups for rules
- `azurerm_firewall_application_rule_collection`: support source_ip_groups for rules
- `azurerm_ip_groups` new resource and data source

Closes: #7047

### CODE REVIEW

File: azurerm/internal/services/network/tests/ip_group_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+}
+
+func testAccAzureRMIpGroup_complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```
Comment (katbyte) [inline azurerm/internal/services/network/tests/ip_group_resource_test.go:222]: we try to put the service/other identifying name in resource groups

```suggestion
  name     = "acctestRG-network-%d"
```

**Review by katbyte — APPROVED** [review]
Aside from one minor comments about resource group names LGTP 👍

File: azurerm/internal/services/network/tests/ip_group_resource_test.go (hunk 1 above)
Comment (manicminer) [reply azurerm/internal/services/network/tests/ip_group_resource_test.go]: Great, done!

**Comment by alexwilcox9** [comment]
Hi, I see this PR is adding a data source as well as a resource but the changelog only mentions the data source
Is this an error in the changelog or will the resource be in a different release?

**Comment by manicminer** [comment]
@alexwilcox9 looks like this is an accidental omission. both the resource and data source should make it in the next release. Good catch, thanks!

**Pull request merged**